File: 2026/08-August/18-Tuesday/solution.java
import java.util.*;

class Solution {
    public int largestInteger(int[] nums, int k) {
        Map<Integer, Integer> freq = new HashMap<>();

        // Count how many windows contain each number
        for (int i = 0; i <= nums.length - k; i++) {
            Set<Integer> seen = new HashSet<>();

            for (int j = i; j < i + k; j++) {
                seen.add(nums[j]);
            }

            for (int x : seen) {
                freq.put(x, freq.getOrDefault(x, 0) + 1);
            }
        }

        int ans = -1;

        // Almost missing = appears in exactly one window
        for (Map.Entry<Integer, Integer> entry : freq.entrySet()) {
            if (entry.getValue() == 1) {
                ans = Math.max(ans, entry.getKey());
            }
        }

        return ans;
    }
}

File: 2026/08-August/18-Tuesday/solution.cpp
class Solution {
public:
    int largestInteger(vector<int>& nums, int k) {
        unordered_map<int, int> freq;
        
        // Count how many distinct windows contain each number
        for (int i = 0; i <= (int)nums.size() - k; i++) {
            unordered_set<int> seen;
            
            for (int j = i; j < i + k; j++) {
                seen.insert(nums[j]);
            }
            
            for (int x : seen) {
                freq[x]++;
            }
        }
        
        int ans = -1;
        
        // Almost missing = appears in exactly one window
        for (auto &[x, count] : freq) {
            if (count == 1) {
                ans = max(ans, x);
            }
        }
        
        return ans;
    }
};
